Cache/IO.lean

```diff
@@ -245,10 +245,13 @@ Each build file also has a `Bool` indicating whether that file is required for c
 def mkBuildPaths (path : FilePath) : IO $ List (FilePath × Bool) := do
   let packageDir ← getPackageDir path
   return [
-    (packageDir / LIBDIR / path.withExtension "trace", true),
+    (packageDir / LIBDIR / path.withExtension "trace", true), -- this one has to be first
```

Comment:
Why does it?

Reply:
it is pattern matched down below, the trace file has a special place in the LTAR format

Reply:
The match in question is:
```
          let trace :: args := (← buildPaths.filterM (·.1.pathExists)) |>.map (·.1.toString)
            | unreachable!
```
Probably this could be written in a way that makes it clearer, but that's orthogonal to this PR.

bors d+

I assume you tested this works?

Reply:
I tested that it is packing / unpacking the .hash files as expected, but I have not reproduced the initial bad cache issue so I can't say I have checked that this fixes it. (But it is very obvious to me that the original implementation would mess lake up and this is a necessary condition for fixing the bug.)

Reply:
The updated version has been tested to work with `lake exe cache get && lake build`.